packages/hooks/use-combined-state/src/useCombinedState.test.tsx

```diff
@@ -131,4 +131,62 @@ describe('useCombinedState', () => {
     expect(isControlledValue).toBe(true)
     expect(response).toBe(789)
   })
+
+  it('value must NOT be updated using its setter when using uncontrolled mode using shouldUpdate flag to false', () => {
+    // Given
+    const args = { value: undefined, defaultValue: 123 } as argsType
+
+    // When
+    const hook = renderHook(() => useCombinedState(...map(args)))
+    let [response, setResponse, isControlledValue, initialValue] = hook.result.current
+
+    // Then
+    expect(typeof response).toBe('number')
```

Comment:
I think you might not need some assertions like this one :) We have type safety using TS + other checks implicitely covering it:
`expect(response).toBe(123)`

Reply:
DONE!